src/types.d.ts

```diff
@@ -0,0 +1,1194 @@
+
+    setMediaAspectRatio(mediaAspectRatio: number): void
+
+    setParameters(params: FlatViewParams): void
+
+    setX(x: number): void
+
+    setY(y: number): void
+
+    setZoom(zoom: number): void
+
+    x(): number
+
+    y(): number
+
+    zoom(): number
+
+    destroy(): void
+
+    height(): number
+
+    intersects(rectangle: [number, number][]): boolean
+
+    inverseProjection(): Array<number>
+
+    projection(): [number, number, number, number]
+
+    selectLevel(levelList: number[]): number
+
+    setSize(size: Size): void
+
+    size(size?: Size): Size
+
+    width(): number
+
+    pitch(): number
+
+    yaw(): number
+  }
+
+  interface FlatViewParams extends Coords {
+    zoom?: number
+    mediaAspectRation: number
+  }
+
+  type FlatViewLimiter = (params: FlatViewParams) => FlatViewParams
+
+  /**
+   * HammerGestures
+   */
+  export abstract class HammerGestures {}
+
+  /**
+   * Hotspot
+   */
+  export abstract class Hotspot {
+    constructor(
+      domElement: HTMLElement,
+      view: View,
+      coords: RectilinearCoords,
+      opts: HotspotOptions
+    )
+
+    destroy(): void
+
+    domElement(): HTMLElement
+
+    perspective(): Perspective
+
+    position(): RectilinearCoords
+
+    setPerspective(perspective: Perspective): void
+
+    setPosition(coords: RectilinearCoords): void
+
+    show(): void
+  }
+
+  export interface HotspotOptions {
+    perspective?: Perspective
+  }
+
+  /**
+   * HotspotContainer
+   */
+  export class HotpsotContainer {
+    constructor(
+      parentDomElement: HTMLElement,
+      stage: Stage,
+      view: View,
+      renderLoop: RenderLoop,
+      opts?: { rect?: RectSpec }
+    )
+
+    createHotspot(
+      domElement: HTMLElement,
+      coords?: RectilinearCoords,
+      opts?: HotspotOptions
+    ): Hotspot
+
+    destroy(): void
+
+    destroyHotspot(hotspot: Hotspot): void
+
+    domElement(): HTMLElement
+
+    hasHotspot(hotspot: Hotspot): boolean
+
+    hide(): void
+
+    listHotspots(): Hotspot[]
+
+    rect(): Rect
+
+    setRect(rect: Rect): void
+
+    show(): void
+  }
+
+  /**
+   * HtmlImageLoader
+   */
+  export class HtmlImageLoader {
+    constructor(stage: Stage)
+
+    loadImage(url: string, rect?: Rect, done?: Function)
+  }
+
+  export interface HtmlImageLoaderInteface extends HtmlImageLoader {}
+
+  /**
+   * ImageUrlSource
+   */
+  export class ImageUrlSource implements Source {
+    constructor(
+      sourceFromTile: (tile: Tile) => { url: string; rect?: Rect },
+      opts?: { concurrency?: number; retryDelay?: number }
+    )
+
+    static fromString: (
+      url: string,
+      options?: {
+        cubeMapPreviewFaceOrder?: string
+        cubeMapPreviewUrl?: string
+        retryDelay?: number
+      }
+    ) => ImageUrlSource
+
+    loadAsset(stage: Stage, tile: Tile, done: Function): Function
+
+    addEventListener(event: Event, callback: Function)
+
+    removeEventListener(event: Event, callback: Function)
```

Comment:
`Function` is rather vague.

We should define custom types for those callbacks. However, those are not documented and I am not sure I get them right when reading the JS code...